contrib/opentimelineio_contrib/adapters/otio_svg.py

```diff
@@ -0,0 +1,481 @@
+green = Color(0.0, 0.5, 0.0, 1.0)
+dark_green = Color(0.0, 0.39, 0.0, 1.0)
+yellow = Color(1.0, 1.0, 0.0, 1.0)
+gold = Color(1.0, 0.84, 0.0, 1.0)
+orange = Color(1.0, 0.647, 0.0, 1.0)
+red = Color(1.0, 0.0, 0.0, 1.0)
+dark_red = Color(0.54, 0.0, 0.0, 1.0)
+brown = Color(0.54, 0.27, 0.1, 1.0)
+pink = Color(1.0, 0.75, 0.79, 1.0)
+gray = Color(0.5, 0.5, 0.5, 1.0)
+dark_gray = Color(0.66, 0.66, 0.66, 1.0)
+
+
+class Point:
+    x = 0.0
+    y = 0.0
+
+    def __init__(self, x=0.0, y=0.0):
+        self.x = x
+        self.y = y
+
+
+class Rect:
+    origin = Point()
+    width = 0.0
+    height = 0.0
+
+    def __init__(self, origin=Point(), width=0.0, height=0.0):
+        self.origin = origin
+        self.width = width
+        self.height = height
+
+    def normalized(self):
+        normalized_origin = Point(
+            self.origin.x + (self.width if self.width < 0 else 0),
+            self.origin.y + (self.height if self.height < 0 else 0),
+        )
+        normalized_width = abs(self.width)
+        normalized_height = abs(self.height)
+        return Rect(normalized_origin, normalized_width, normalized_height)
+
+    def min_x(self):
+        return self.normalized().origin.x
+
+    def min_y(self):
+        return self.normalized().origin.y
+
+    def mid_x(self):
+        return self.origin.x + (self.width / 2)
+
+    def mid_y(self):
+        return self.origin.y + (self.height / 2)
+
+    def max_x(self):
+        norm = self.normalized()
+        return norm.origin.x + norm.width
+
+    def max_y(self):
+        norm = self.normalized()
+        return norm.origin.y + norm.height
+
+    def contract(self, distance):
+        self.origin.x += distance
+        self.origin.y += distance
+        self.width -= 2 * distance
+        self.height -= 2 * distance
+
+
+class SVGRenderer:
+    LCARS_CHAR_SIZE_ARRAY = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                             0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                             0, 0, 17, 26, 46, 63, 42, 105, 45, 20, 25, 25, 47, 39, 21,
+                             34, 26, 36, 36, 28, 36, 36, 36,
+                             36, 36, 36, 36, 36, 27, 27, 36, 35, 36, 35, 65, 42, 43, 42,
+                             44, 35, 34, 43, 46, 25, 39, 40,
+                             31, 59, 47, 43, 41, 43, 44, 39, 28, 44, 43, 65, 37, 39, 34,
+                             37, 42, 37, 50, 37, 32, 43, 43,
+                             39, 43, 40, 30, 42, 45, 23, 25, 39, 23, 67, 45, 41, 43, 42,
+                             30, 40, 28, 45, 33, 52, 33, 36,
+                             31, 39, 26, 39, 55]
+    width = 1000.0
+    height = 660.0
+    font_family = "Roboto"
+    lines = ['']
+
+    def __init__(self, width=1000.0, height=660.0, font_family="Roboto"):
+        self.width = width
+        self.height = height
+        self.font_family = font_family
+
+    def convert_point_to_svg_coordinates(self, point=Point()):
+        y = self.height - point.y
```

Comment:
Actually I ported a renderer I wrote for my [data visualization library](https://github.com/KarthikRIyer/swiftplot) to python. There I had to do this conversion.

Yes, we could store stuff y-down. But I found it a bit easier to think about the drawing algorithm in y-up because that's the way we generally think of when it comes to plotting  stuff.

Reply:
Makes sense. There are also well known graphics systems that use y-down coordinates, so  as long as its documented I wouldn't worry about future developers understanding what that means and how to work with it.(for example: https://renderman.pixar.com/resources/RenderMan_20/appnote.10.html).  

Related: I forget if I already suggested it, but it would be good to document the coordinate systems values are stored in somewhere (something as simple as "points are y-down").

Reply:
Would mentioning it with the title `"""OTIO to SVG Adapter"""` of the adapter be enough?

Reply:
You mean in the top level doctoring?  Sure!